tests/cupyx_tests/scipy_tests/interpolate_tests/test_ppoly.py

```diff
@@ -0,0 +1,748 @@
+
+        # 'scalars'
+        p = cls(c[..., 0, 0, 0], x)
+
+        assert p(0.5).shape == ()
+        assert p(xp.array(0.5)).shape == ()
+
+        xxx = xp.array([[0.1, 0.2], [0.4]], dtype=object)
+        p(xxx)   # raises ValueError
+
+    @parametrize_cls
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_complex_coef(self, xp, scp, cls):
+        cls = getattr(scp.interpolate, cls)
+        numpy.random.seed(12345)
+        x = numpy.sort(numpy.random.random(13))
+        x = xp.array(x)
+        c = numpy.random.random((8, 12)) * (1. + 0.3j)
+        c = xp.array(c)
+        # c_re, c_im = c.real, c.imag
+        xpt = xp.array(numpy.random.random(5))
+
+        p = cls(c, x)
+        return [p(xpt, nu) for nu in [0, 1, 2]]
+
+    @parametrize_cls
+    @pytest.mark.parametrize('axis', [0, 1, 2, 3])
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_axis(self, xp, scp, cls, axis):
+        cls = getattr(scp.interpolate, cls)
+        numpy.random.seed(12345)
+        c = numpy.random.rand(3, 4, 5, 6, 7, 8)
+        c = xp.asarray(c)
+        c_s = c.shape
+        xpt = numpy.random.random((1, 2))
+        xpt = xp.asarray(xpt)
+
+        m = c.shape[axis+1]
+        x = numpy.sort(numpy.random.rand(m+1))
+        x = xp.asarray(x)
+
+        p = cls(c, x, axis=axis)
+        assert (p.c.shape ==
+                c_s[axis:axis+2] + c_s[:axis] + c_s[axis+2:])
+
+        res = p(xpt)
+        targ_shape = c_s[:axis] + xpt.shape + c_s[2+axis:]
+        assert res.shape == targ_shape
+
+        # deriv/antideriv does not drop the axis
+        for p1 in [cls(c, x, axis=axis).derivative(),
+                   cls(c, x, axis=axis).derivative(2),
+                   cls(c, x, axis=axis).antiderivative(),
+                   cls(c, x, axis=axis).antiderivative(2)]:
+            assert p1.axis == p.axis
+
+        return True
+
+    @parametrize_cls
+    @pytest.mark.parametrize('axis', [-1, 4, 5, 6])
+    @testing.numpy_cupy_allclose(scipy_name='scp', accept_error=ValueError)
+    def test_axis_2(self, xp, scp, cls, axis):
+        cls = getattr(scp.interpolate, cls)
+        numpy.random.seed(12345)
+        c = numpy.random.rand(3, 4, 5, 6, 7, 8)
+        c = xp.asarray(c)
+
+        x = numpy.sort(numpy.random.rand(c.shape[-1]))
+        x = xp.asarray(x)
+
+        # c array needs two axes for the coefficients and intervals, so
+        # 0 <= axis < c.ndim-1; raise otherwise
+        instance = cls(c=c, x=x, axis=axis)
+        return instance.c, instance.x, instance.axis
+
+
+@testing.with_requires("scipy")
+class TestPPoly:
+    @testing.numpy_cupy_allclose(scipy_name='scp')
+    def test_simple(self, xp, scp):
+        c = xp.array([[1, 4], [2, 5], [3, 6]])
+        x = xp.array([0, 0.5, 1])
+        p = scp.interpolate.PPoly(c, x)
+        testing.assert_allclose(p(0.3), 1*0.3**2 + 2*0.3 + 3)
+        testing.assert_allclose(p(0.7), 4*(0.7-0.5)**2 + 5*(0.7-0.5) + 6)
+        return True
```

Comment:
Does the following work with `numpy_cupy_allclose`?
```suggestion
        return [p(0.3), p(0.7)]
```